src/services/urlUtils.ts

```diff
@@ -122,41 +122,53 @@ namespace UrlUtils {
     });
   };
 
-  export const getNavigations = (request: UrlBeautifier.SearchUrlState) => {
+  // tslint:disable-next-line max-line-length
+  export const getNavigations = (state: Store.Indexed<Store.Navigation> | Store.AvailableNavigations, request: UrlBeautifier.SearchUrlState) => {
     return {
-      allIds: getAllIds(request),
-      byId: getById(request),
+      allIds: getAllIds(state, request),
+      byId: getById(state, request),
     };
   };
 
-  export const getAllIds = (request: UrlBeautifier.SearchUrlState) => {
-    return request.refinements.reduce((fields, refinement) => {
-      if (!fields.includes(refinement.field)) {
-        fields.push(refinement.field);
-      }
-      return fields;
-    }, []);
+  // tslint:disable-next-line max-line-length
+  export const getAllIds = (state: Store.Indexed<Store.Navigation> | Store.AvailableNavigations, request: UrlBeautifier.SearchUrlState) => {
+    const allIds = [...state.allIds];
+    allIds.push(...request.refinements.filter(({ field }) => !allIds.includes(field)).map(({ field }) => field));
```

Comment:
I would `map` first then `filter` so you don't destructure twice.

Reply:
I realized that instead of spreading to `push`, it would be better to just use `concat`.